### ISSUE TITLE
Check whether negative integer literals are out of range for their types

### ISSUE THREAD

**Issue by michaeljklein** [noir-lang/noir#6247]
### Problem

The following test checks that a literal `-1` is out of range for `u8`, and currently fails:

```rust
#[test]
fn cast_negative_one_to_u8_size_checks() {
    let src = r#"
        fn main() {
            assert((-1) as u8 != 0);
        }
    "#;
    let errors = get_program_errors(src);
    assert!(!errors.is_empty());
}
```

### Happy Case

Reject negative literals outside of their type's range
- Ensure that `-1` as a `u8` fails with an error
- Ensure that using `i8`, `i16`, .. negative literals outside of their type's range fails with an error

### Workaround

None

### Project Impact

None

### Would you like to submit a PR for this Issue?

None

**Comment by michaeljklein** [comment]
Once https://github.com/noir-lang/noir/pull/6137 is merged and this is fixed, https://github.com/noir-lang/noir/issues/6219 will be addressed.

Noting that `(0 - 1) as u8` is `wontfix`
